### ISSUE TITLE
[BUG] Calico strictAffinity is false when installing Windows Node

### ISSUE THREAD

**Issue by Danielku15** [Azure/AKS-Edge#43]
**Describe the bug**
I followed https://learn.microsoft.com/en-us/azure/aks/hybrid/aks-edge-howto-single-node-deployment with also adding a Windows Node. When I tried to deploy windows pods or follow https://learn.microsoft.com/en-us/azure/aks/hybrid/aks-edge-howto-use-storage-local-path the pods do not come up due to errors like: 

> networkPlugin cni failed to set up pod "kube-proxy-windows-zc9mt_kube-system" network: global strict affinity should not be false for Windows node

>   Warning  FailedCreatePodSandBox  2m57s (x645 over 17m)  kubelet            (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "b8bf687e6209c3a84f5ad62387d335c991d2f4bbe717b836e1e0e03ba2b68f96" network for pod "local-path-provisioner-86f4f99d4c-xcxzd": networkPlugin cni failed to set up pod "local-path-provisioner-86f4f99d4c-xcxzd_kube-system" network: global strict affinity should not be false for Windows node

**To Reproduce**
1. Follow the https://learn.microsoft.com/en-us/azure/aks/hybrid/aks-edge-howto-single-node-deployment guide
2. Follow the https://learn.microsoft.com/en-us/azure/aks/hybrid/aks-edge-howto-use-storage-local-path
3. `kubectl get pods --namespace=kube-system` -> Notice that some containers are stuck in `ContainerCreating` (e.g. kube-proxy-windows-#### and local-path-provisioner-###)
4. `kubectl describe pod kube-proxy-windows --namespace=kube-system` -> Notice the error

**Expected behavior**
When setting up AKS Edge with windows nodes and calico, the strict affinity should be configured accordingly

**Screenshots**
--

**Environment (please complete the following information):**
 - AKS Edge Essentials Version AksEdge-K8s-0.7.22335.1024.msi
 - Kubernetes version 1.22
 - Windows Host OS (please complete the following information):
    - Edition: Windows Server 2019 Datacenter 
    - Version: Build 17763
    - Virtual Machine: Root Server

**Additional context**
I tried to set the strictAffinity to true via guides like 
https://projectcalico.docs.tigera.io/getting-started/windows-calico/quickstart#configure-strict-affinity-for-clusters-using-calico-networking but I somehow failed. A command like the following also fails with this error: 

```
C:\AKS-Edge>kubectl calico ipam configure --strictaffinity=true --config=calico.cfg.yaml --allow-version-mismatch
Error: global strict affinity should not be false for Windows node
Error: exit status 1
C:\AKS-Edge>kubectl calico ipam show --config=calico.cfg.yaml --allow-version-mismatch
+----------+---------------+-----------+------------+--------------+
| GROUPING |     CIDR      | IPS TOTAL | IPS IN USE |   IPS FREE   |
+----------+---------------+-----------+------------+--------------+
| IP Pool  | 10.244.0.0/16 |     65536 | 7 (0%)     | 65529 (100%) |
+----------+---------------+-----------+------------+--------------+

C:\AKS-Edge>kubectl calico ipam show --show-configuration --config=calico.cfg.yaml --allow-version-mismatch
Error: global strict affinity should not be false for Windows node
Error: exit status 1
```

Would be great to get some info how I can change the strict affinity manually without running into the same error. 

This issue might be related: https://github.com/projectcalico/calico/issues/4404

**Comment by pbering** [comment]
@Danielku15 I have the same problem, possible workaround that worked for me was to first install only the Linux node and when finished then add Windows node with `Add-AksEdgeNode -NodeType Windows -Ip4Address "x.x.x.x" -CpuCount 8 -MemoryInMB 16384`. 

Also the Caclico quickstart states that "_For **Linux control nodes using Calico** networking, strict affinity must be set to true. This is required to prevent Linux nodes from borrowing IP addresses from Windows nodes_" so we need to find a way to set `strictAffinity` to `true` on the Linux node and NOT the Windows node! Anyone knows how to do that?

**Comment by Danielku15** [comment]
I managed it last week but forgot to update here. It really seems that the `--allow-version-mismatch` in my case was to blame for the error. In AKS Edge there is calico `3.20` but I couldn't find a `3.20` executable of calicoctl in the releases. So what I did was the following: 

1. I pulled the calico docker image in the exact version running: https://projectcalico.docs.tigera.io/maintenance/clis/calicoctl/install#install-calicoctl-as-a-container-on-a-single-host in 
2. I used `docker run` to run exactly this image with the correct command `calicoctl ipam configure --strictaffinity=true`. 
3. To get it fully running I had to fiddle a bit by mounting a kube/config and setting the environment variables correct. I ended up with something like: `docker run -e "DATASTORE_TYPE=kubernetes" -e "KUBECONFIG=/kube/config" -v C:\users\daniel\.kube:/kube calicoctl ipam configure --strictaffinity=true`

the command was successful and the errors disappeared. Windows node and pods were already configured in the cluster. Some killing/restart of pods might be needed.

**Comment by pbering** [comment]
Nice! I will test that approach out. I hope that in the future the `New-AksEdgeDeployment` will handle it automatically when the `NodeType` is `LinuxAndWindows` and network is Calico :)

**Comment by fcabrera23** [comment]
Hi @pbering and @Danielku15,

Thanks for your detailed information and the workarounds provided. We are fixing this issue in our next release. 

Thanks,
Francisco

**Comment by fcabrera23** [comment]
Hi @pbering & @Danielku15,

This issue was fixed in the [1.0.266.0](https://github.com/Azure/AKS-Edge/releases/tag/1.0.266.0) update.

Thanks,
Francisco
